### PR TITLE
Fixes final stop message handling #26141

### PR DESCRIPTION
**Pull request by nvollmar**
Adds a test and fix for Issue #26141

### CODE REVIEW

**Comment by patriknw** [comment]
OK TO TEST

File: akka-actor/src/main/scala/akka/pattern/BackoffSupervisor.scala [hunk 1]
```diff
@@ -436,8 +436,9 @@ private[akka] trait HandleBackoff { this: Actor ⇒
         finalStopMessage match {
           case None ⇒
           case Some(fsm) ⇒
-            fsm(msg)
-            context.stop(self)
+            if (fsm(msg)) {
+              context.stop(self)
+            }
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/pattern/BackoffSupervisor.scala]: great catch, thanks

**Review by patriknw — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]

**Comment by nvollmar** [comment]
@patriknw I can't access Jenkins to see what's wrong, local sbt validatePullRequest was fine. And Jenkins says 0 tests failed regardless? `Test FAILed. 8103 tests run, 498 skipped, 0 failed`

**Comment by raboof** [comment]
@nvollmar are you sure you cannot access jenkins? if so then that isn't intentional and we'd like to fix it. Clicking on 'Details' next to 'Jenkins PR Validation' should bring you to https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/14308/, where 'Console Output' (on the left) should show the logs. Seems to work from an incognito window for me.

**Comment by raboof** [comment]
The error seems to be a parboiled timeout while generating the docs with 'sbt paradox' - I'm not sure I've ever seen that before, perhaps it was a hardware hiccup.
```
[12-20 07:49:31.078] [error] org.parboiled.errors.ParserRuntimeException: Error while parsing action 'Root/Sequence/ZeroOrMore/Sequence/Block/FirstOf/Sequence/FirstOf/Para/Sequence/Inlines/Sequence/ZeroOrMore/Sequence/InlineOrIntermediateEndline/Sequence/Inline/Inline_Action1' at input position (line 166, pos 74):
[12-20 07:49:31.078] [error] attaching it to our `factorials` source—after a small adaptation to turn the
[12-20 07:49:31.078] [error]                                                                          ^
[12-20 07:49:31.078] [error] 
[12-20 07:49:31.078] [error] org.pegdown.ParsingTimeoutException
[12-20 07:49:31.078] [error] 	at org.parboiled.MatcherContext.runMatcher(MatcherContext.java:367)
[12-20 07:49:31.078] [error] 	at org.parboiled.matchers.SequenceMatcher.match(SequenceMatcher.java:46)
```
trying again.

**Comment by raboof** [comment]
PLS BUILD

**Comment by nvollmar** [comment]
@raboof The reason is that Jenkins is on a non-standard port that is blocked in our office

**Comment by raboof** [comment]
@nvollmar aah, I've run into that myself before as well. Let's create a ticket, though it's not very a very high priority. Added #26143

**Review by chbatey — APPROVED** [review]

**Pull request merged**